ibis/backends/duckdb/tests/test_register.py

```diff
@@ -350,3 +350,20 @@ def test_csv_with_slash_n_null(con, tmp_path):
     t = con.read_csv(data_path, nullstr="\\N")
     col = t.a.execute()
     assert pd.isna(col.iat[-1])
+
+
+@pytest.mark.xfail(
+    LINUX and SANDBOXED,
+    reason=("nix can't hit GCS because it is sandboxed."),
+)
+def test_register_filesystem_gcs(con):
+    import fsspec
+
+    gcs = fsspec.filesystem("gcs")
+
+    con.register_filesystem(gcs)
+    band_members = con.read_csv(
```

Comment:
Is there a way we can make the `con.register_filesystem` call automatic for the names in the allowed list?

Reply:
Right now we could do that for `gcs`, but for `azure` or `s3` it would be ambiguous whether a user wanted to use fsspec or a native DuckDB extension.
I don't _think_ we want to have it auto register for only some filesystem-like things. 
We might be able to design a consistent set of fallback behaviors?

Reply:
Nah, it's fine as is. Perhaps a follow up with documentation about what to do in various situations? Ideally we don't want to have different ways to do the very similar things.